File: Ejercicios Propuestos 1/Ejercicio 14/Ejercicio 14.jl
suma = 0
cantidad = 0
menor = 0
mayor = 0

print("Ingrese números enteros positivos (Escriba 0 para terminar): ")
num = parse(Int, readline())

while num != 0
    if num < 0
        println("Error: Solo se permiten números positivos.")
    else
        global suma += num
        global cantidad += 1
        
        if cantidad == 1
            global menor = num
            global mayor = num
        else
            if num < menor
                global menor = num
            end
            if num > mayor
                global mayor = num
            end
        end
    end
    print("Ingrese número: ")
    global num = parse(Int, readline())
end

if cantidad > 0
    promedio = suma / cantidad
    println("El número MENOR es: ", menor)
    println("El número MAYOR es: ", mayor)
    println("El PROMEDIO es: ", promedio)
else
    println("No se ingresaron números válidos.")
end

File: Ejercicios Propuestos 1/Ejercicio 14/Ejercicio14.java
import java.util.Scanner;
public class Ejercicio14 {
    public static void main(String[] args) {
        try (Scanner sc = new Scanner(System.in)) {
            int num, mayor = Integer.MIN_VALUE, menor = Integer.MAX_VALUE;
            int suma = 0, contador = 0;
            System.out.println("Ingrese números (0 para finalizar):");
            while (true) {
                num = sc.nextInt();
                if (num == 0) break;
                if (num > mayor) mayor = num;
                if (num < menor) menor = num;
                suma += num;
                contador++;
            }
            if (contador > 0) {
                System.out.println("Mayor: " + mayor);
                System.out.println("Menor: " + menor);
                System.out.println("Media: " + (double) suma / contador);
            } else {
                System.out.println("No se ingresaron números.");
            }
        }
    }
}